lib/biran/configurinator.rb

```diff
@@ -103,10 +103,10 @@ def local_config_file_contents
       @local_config_contents = process_config_file(local_config_file)
     end
 
-    def get_secret_contents(app_config)
+    def get_secrets_content(secrets_file)
       secrets_file_contents = {}
-      if File.exist? app_config[:secrets_file_path]
-        secrets_file_contents = process_config_file app_config[:secrets_file_path]
+      if File.exist? secrets_file
+        secrets_file_contents = process_config_file secrets_file
       end
       secrets_file_contents
```

Comment:
I mostly don't use ternary, but I think here is a good candidate

```
def get_secrets_content(secrets_file)
  File.exist?(secrets_file) ? process_config_file(secrets_file) : {}
end
```

or you can use early return:

```
def get_secrets_content(secrets_file)
  return {} unless File.exist?(secrets_file)
  process_config_file(secrets_file)
end
```

Reply:
another option is to let it fail and rescue on Missing config file

```
def get_secrets_content(secrets_file)
  process_config_file(secrets_file)
rescue RuntimeError => error
  return {} if error.message =~ /Missing config file/i
end
```

I think the last one is the best option... but will need to move and re raise a more expressive error on the process config file method

Reply:
I will look at the early return option, but the file is optional, so we don't want to fail on it missing. In this case, I like the early return over the ternary as I think it might be a little more clear.

Reply:
With the rescue you avoid the failure, but yeah it is fine to check for the files existence. Now that I think twice, it is better.

Reply:
right, we just need to see if it is there to use it. If it is not there, we don't care. In reality, there should be no required file for biran to work, however, I think `config/app_config.yml` is still a little bit required as we may not have enough defaults defined. If this was a required file 100% a rescue would be a good idea.